rasa/nlu/utils/bilou_utils.py

```diff
@@ -240,22 +244,72 @@ def _add_bilou_tags_to_entities(
                 bilou[end_token_idx] = f"L-{label}"
 
 
-def _get_entity_positions(entities: List[Tuple[int, int, Text]]) -> Set[int]:
-    entity_positions = set()
+def check_consistent_bilou_tagging(predicted_tags: List[Text]) -> None:
+    """
+    Check if predicted tags follow the BILOU tagging schema.
 
-    for start_pos, end_pos, label in entities:
-        for i in range(start_pos, end_pos):
-            entity_positions.add(i)
+    Args:
+        predicted_tags: list of predicted tags
+    """
 
-    return entity_positions
+    # TODO
+    #  we might want to correct cases like B-location I-person L-location ->
+    #  B-location I-location L-location instead of just logging warnings
 
+    last_tag = NO_ENTITY_TAG
+    entity_found = False
 
-def _handle_not_an_entity(
-    bilou: List[Text], tokens: List[Token], entity_positions: Set[int], missing: Text
-):
-    for n, token in enumerate(tokens):
-        for i in range(token.start, token.end):
-            if i in entity_positions:
-                break
-        else:
-            bilou[n] = missing
+    for predicted_tag in predicted_tags:
+        prefix = bilou_prefix_from_tag(predicted_tag)
+        tag = tag_without_prefix(predicted_tag)
+
+        if not prefix:
+            if entity_found:
+                logger.debug(
+                    f"Inconsistent BILOU tagging schema found. Missing "
+                    f"'L-{last_tag}'."
+                )
+
+            entity_found = False
+            last_tag = NO_ENTITY_TAG
+            continue
+
+        if prefix == "B":
```

Comment:
why are these prefixes different from the ones listed in `BILOU_PREFIXES` (without the `-` suffix)?

Reply:
what do you think about creating a `NamedTuple` and accessing the prefixes with `nt.beginning`, `nt.inside` etc? this could replace the `BILOU_PREFIXES` at the top and be used across this file